### ISSUE TITLE
[FR] Parallel Coordinates Plot  - Select/Deselect all parameters/metrics

### ISSUE THREAD

**Issue by Bibonaut** [mlflow/mlflow#2273]
Thank you for submitting an issue. Please refer to our [issue policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)
for information on what types of issues we address.

Please fill in this template and do not delete it unless you are sure your issue is outside its scope.

-------
## Describe the proposal
When comparing runs with Parallel Coordinates Plot it shows all parameters by default. There is no option to select or deselect all parameters or metrics.  A button to select/deselect all parameters/metrics should be added.

### Motivation
Most experiments/runs have many parameters, but only few are changed between runs or are relevant for run comparison analysis. Since all parameters are enabled in Parallel Coordinates Plot by default, it costs much time to deselect irrelevant parameters. Furthermore, having many parameters enabled in Parallel Coordinates Plots, the UI is less responsive, which increases time further to set up the plot as wanted. 

### Proposed Changes
Add buttons for parameters and metrics to toggle selection of parameters/metrics.

**Comment by harupy** [comment]
Like this?
![toggle-all](https://user-images.githubusercontent.com/17039389/71962686-53e2fc80-323d-11ea-9a4c-d058413f0936.gif)

**Comment by Bibonaut** [comment]
@harupy Looks very nice! That's exactly what I need. I am running v1.5.0. (Windows, if this matters) 
How can I get it? :)

**Comment by harupy** [comment]
@Bibonaut 
I think I need to open a pull request.

If you want to use this right now, please run the following commands which should enable this feature.
```
# clone my fork containing this feature
git clone https://github.com/harupy/mlflow/tree/toggle-all

# move to the UI directory
cd mlflow/mlflow/server/js

# install JS dependencies
npm install

# build package
npm run build

# Move to the root and install mlflow
cd ../../..
pip install -e .
```

**Comment by harupy** [comment]
cc @gioa @smurching @Zangr 

What do you think about this change?

**Comment by gioa** [comment]
@harupy This is great! Thank you for working on it!

As @Bibonaut mentioned, in most cases, only different parameters are relevant for run comparison analysis, I am wondering whether it makes sense to add a "Select diff only" as well. See the attached mock. Thoughts? @smurching @Zangr @mateiz 

![ShowDiffOnly](https://user-images.githubusercontent.com/1960721/72011268-4e21f280-320e-11ea-944e-cf9e225eaf2e.png)

**Comment by harupy** [comment]
@gioa Thanks for the feedback. "Select diff only" is a good idea! It seems a bit complicated to implement though.

**Comment by gioa** [comment]
I understand. Let's leave it to another PR then. Thank you for considering it! @harupy

**Comment by pogil** [comment]
@harupy did you wind up submitting a pull request for this? I do think this would be valuable even without the diff only option @gioa suggested.

**Comment by harupy** [comment]
@pogil No, but I have a branch that I created to capture the demo in https://github.com/mlflow/mlflow/issues/2273#issuecomment-571945292. I'll clean up the code and submit a pull request.

**Comment by Bibonaut** [comment]
It would be very nice, if only a few or a maximum of `N` parameters are selected by default (I think this is how it is done in tensorboard).
I had problems to use the parallel coordinates plot because I have so many parameters and when I open the dialog, everything gets stuck and I have to close the web page.

**Comment by Zangr** [comment]
Currently, we automatically select all parameters and metrics that are present in all comparing runs, i.e. if paramA exists in all runs but one, it will be dropped. We can still select paramA afterwards but the line for run without paramA will be dropped from the plot.
```
export const getSharedParamKeysByRunUuids = (runUuids, state) =>
  _.intersection(
    ...runUuids.map((runUuid) => Object.keys(state.entities.paramsByRunUuid[runUuid])),
  );
```

**Comment by dbczumar** [comment]
Parameter selection for the parallel coordinates plot is now much simpler and easier to control. Thank you for using MLflow!
